main.php

```diff
@@ -71,6 +70,21 @@
 		exit(1);
 	}
 }
+// Allow a first class env var to be set that disables .env file loading
+if (!getenv('SS_IGNORE_DOT_ENV')) {
```

Comment:
Why has this moved from Constants.php to main.php? It's a lot less DRY. You've also forgotten to include it in cli-script.php.

My recommendation would be to shift the env load back to Constants.php after BASE_PATH is defined.